portfolio/src/main/java/com/google/sps/servlets/ImageBlobsServlet.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Serves image blobs directly from the blobstore */
+@WebServlet("/serve-images")
+public class ImageBlobsServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String queryString = request.getQueryString();
+    if (queryString == null) {
+      return;
+    }
+    BlobKey blobKey = new BlobKey(request.getParameter("blob-key"));
```

Comment:
Have you thought about what happens if the user gives an invalid blob-key, or doesn't provide one?

Reply:
I did think about these cases before, but I knew that my current front-end code checks for whether there is a BlobKey before passing it to the server (hence there will be no null BlobKeys being passed to server). And I knew that invalid BlobKeys will be a 404 error, and to the user it will just look like there wasn't an image submitted in the first place (of course, with errors in the console).

But I guess it's best practice to handle edge cases because I don't want unexpected behaviour later on if more code was to use this servlet 😅 so I've handled those cases now

Reply:
Thanks for handling!